File: repos_estudiantes_TARQUI_primera_entrega/RUBIN_DE_CELIS_VALDEZ_ANDRES/INTRODUCCION-POO/EJERCICIO-9/Computadora.java
public class Computadora 
{
    private String marca;
    private String procesador;
    private int ram; 
    private boolean encendida;

    public Computadora(String marca, String procesador, int ram) 
    {
        this.marca = marca;
        this.procesador = procesador;
        this.ram = ram;
        this.encendida = false;
    }
    
    public void encender() 
    {
        if (encendida) 
        {
            System.out.println("La computadora ya está encendida.");
        } 
        else 
        {
            encendida = true;
            System.out.println("La computadora se ha encendido.");
        }
    }
    
    public void apagar() 
    {
        if (!encendida) 
        {
            System.out.println("La computadora ya está apagada.");
        } 
        else 
        {
            encendida = false;
            System.out.println("La computadora se ha apagado.");
        }
    }
    
    public void mostrarComponentes() 
    {
        System.out.println("Marca: " + marca);
        System.out.println("Procesador: " + procesador);
        System.out.println("RAM: " + ram + " GB");
    }

    public void mostrarEstado() 
    {
        String estado = encendida ? "Encendida" : "Apagada";
        System.out.println("Estado de la computadora: " + estado);
    }
    
    public static void main(String[] args) 
    {
        Computadora miComputadora = new Computadora("Dell", "Intel i7", 16);

        
        System.out.println("Componentes de la computadora:");
        miComputadora.mostrarComponentes();
        System.out.println("-----------------------------");

        
        miComputadora.mostrarEstado();
        System.out.println("-----------------------------");

        
        miComputadora.encender();
        miComputadora.mostrarEstado();
        System.out.println("-----------------------------");

        
        miComputadora.apagar();
        miComputadora.mostrarEstado();
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/CALLE_CASTILLO_CRISTHIAN_JAVIER/PRACTICA1-PROGRAMACION-II/PRACTICA INF-121/PROGRAMACION ORIENTADA A OBJETOS/EJERCICIO 1/Computadora.py
class Computadora:
    def __init__(self, marca, modelo, componentes):
        self.marca = marca
        self.modelo = modelo
        self.componentes = componentes  
        self.estado = False  

    def encender(self):
        self.estado = True
        print("La computadora ha sido encendida.")

    def apagar(self):
        self.estado = False
        print("La computadora ha sido apagada.")

    def mostrar_estado(self):
        estado_str = "Encendida" if self.estado else "Apagada"
        print(f"Estado: {estado_str}")

    def mostrar_componentes(self):
        print("Componentes de la computadora:")
        for comp in self.componentes:
            print(f"- {comp}")

componentes_pc = ["Procesador Intel i7", "8GB RAM", "SSD 512GB", "Tarjeta gráfica NVIDIA", "Placa madre ASUS"]

mi_pc = Computadora("Lenovo", "ThinkPad", componentes_pc)

mi_pc.mostrar_componentes()
mi_pc.mostrar_estado()
mi_pc.encender()
mi_pc.mostrar_estado()
mi_pc.apagar()
mi_pc.mostrar_estado()

